ui/src/plugins/dev.perfetto.AndroidStartup/slowInsights.ts

```diff
@@ -0,0 +1,747 @@
+    `
+    SELECT RUN_METRIC('android/startup/slow_start_thresholds.sql');
+    SELECT threshold_open_dex_files_from_oat_percentage() as threshold_pct;
+    `,
+    tag,
+  );
+  let threshold_pct = 0;
+  let it = result.iter({threshold_pct: LONG});
+  threshold_pct = Number(it.threshold_pct);
+
+  let slices = await getMainThreadSlices(trace, tag, slice_glob, 0, threshold_pct, startups);
+
+  const uri = `/android_startups/open_dex_files_from_oat`;
+  const title = `Open Dex Files From Oat`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getVerifyClassTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'VerifyClass';
+  let slice_glob = `'VerifyClass*'`;
+  let result = await trace.engine.query(
+    `
+    SELECT RUN_METRIC('android/startup/slow_start_thresholds.sql');
+    SELECT threshold_verify_classes_percentage() as threshold_pct;
+    `,
+    tag,
+  );
+  let threshold_pct = 0;
+  let it = result.iter({threshold_pct: LONG});
+  threshold_pct = Number(it.threshold_pct);
+
+  let slices = await getMainThreadSlices(trace, tag, slice_glob, 0, threshold_pct, startups);
+
+  const uri = `/android_startups/verify_class`;
+  const title = `Verify Class`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getLockContentionTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'LockContention';
+  let slice_glob = `'Lock contention on*'`;
+  let result = await trace.engine.query(
+    `
+    SELECT RUN_METRIC('android/startup/slow_start_thresholds.sql');
+    SELECT threshold_lock_contention_percentage() as threshold_pct;
+    `,
+    tag,
+  );
+  let threshold_pct = 0;
+  let it = result.iter({threshold_pct: LONG});
+  threshold_pct = Number(it.threshold_pct);
+
+  let slices = await getMainThreadSlices(trace, tag, slice_glob, 0, threshold_pct, startups);
+
+  const uri = `/android_startups/lock_contention`;
+  const title = `Lock Contention`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getMonitorContentionTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'MonitorContention';
+  let slice_glob = `'Lock contention on a monitor*'`;
+  let result = await trace.engine.query(
+    `
+    SELECT RUN_METRIC('android/startup/slow_start_thresholds.sql');
+    SELECT threshold_monitor_contention_percentage() as threshold_pct;
+    `,
+    tag,
+  );
+  let threshold_pct = 0;
+  let it = result.iter({threshold_pct: LONG});
+  threshold_pct = Number(it.threshold_pct);
+
+  let slices = await getMainThreadSlices(trace, tag, slice_glob, 0, threshold_pct, startups);
+
+  const uri = `/android_startups/monitor_contention`;
+  const title = `Monitor Contention`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getBindApplicationTrack(
```

Comment:
In general there's a lot of duplicated code in this file that makes it very difficult to understand exactly what you're trying to do. Could you remove some of this duplication please?